### PR TITLE
Typescript typings

### PR DESCRIPTION
**Pull request by mmarinovic**
This PR introduces Typescript typings. Let me know if I overlooked something. 

Custom option type:
data-type

Types are infered for other methods which use option:
method-type-infer

### CODE REVIEW

**Comment by mikaello** [comment]
This looks really great! And nice with all the documentation.Two questions / remarks:

* Could the 'data' part of the default documentation be markdown-rendered as code? E.g. ``Default is {}`` -> ``Default is `{}` ``
* Could everywhere you have typed `key` to be `any` be typed as `React.Attributes["key"]` (this is `IOption.key`, `keyExtractor` and `selectedKey`)? I don't know if this is normal to do, but the `key` is passed forward to React-components as `key`-prop so anything other than string/number/undefined is not allowed.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+   * Optional component overrides label text
+   *
+   * Default is []
+   */
+  data: TOption[],
+  
+  /**
+   * Callback function, when the users has selected an option
+   *
+   * Default is () => {}
```
Comment (mikaello) [inline index.d.ts]: Don't know if this is necessary, but could be nice to markdown this as ``Default is `() => {}` ``

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,331 @@
+import React from 'react';
+import { ViewStyle, TextStyle } from 'react-native';
+
+export interface IOption {
+  key: any;
```
Comment (mikaello) [inline index.d.ts]: This is passed forward to React, so the type is really `React.Attributes["key"]`, or alternatively `string | number | undefined`

File: index.d.ts (hunk 1 above)
Comment (mmarinovic) [reply index.d.ts]: Yes, that's nice. I'll add it

File: index.d.ts (hunk 2 above)
Comment (mmarinovic) [reply index.d.ts]: You are right, I'll put `React.Key` here

**Comment by mikaello** [comment]
Thanks @mmarinovic !

**Comment by mmarinovic** [comment]
@mikaello Thanks!

**Pull request merged**